libr/bin/format/pe/pe.c

```diff
@@ -3123,19 +3121,32 @@ struct r_bin_pe_export_t* PE_(r_bin_pe_get_exports)(struct PE_(r_bin_pe_obj_t)*
 			free (func_rvas);
 			return NULL;
 		}
-		r_buf_read_at (bin->b, ordinals_paddr, (ut8 *)ordinals, names_sz);
-		r_buf_read_at (bin->b, functions_paddr, (ut8 *)func_rvas, funcs_sz);
+		int r = r_buf_read_at (bin->b, ordinals_paddr, (ut8 *)ordinals, names_sz);
+		if (r != names_sz) {
+			free (exports);
+			free (ordinals);
+			free (func_rvas);
+			return NULL;
+		}
+		int s = r_buf_read_at (bin->b, functions_paddr, (ut8 *)func_rvas, funcs_sz);
+		if (s != names_sz) {
+			free (exports);
+			free (ordinals);
+			free (func_rvas);
+			return NULL;
+		}
 		for (i = 0; i < bin->export_directory->NumberOfFunctions; i++) {
 			// get vaddr from AddressOfFunctions array
 			function_rva = r_read_at_ble32 ((ut8 *)func_rvas, i * sizeof (PE_VWord), bin->endian);
 			// have exports by name?
-			if (bin->export_directory->NumberOfNames != 0) {
+			if (bin->export_directory->NumberOfNames > 0) {
 				// search for value of i into AddressOfOrdinals
 				name_vaddr = 0;
 				for (n = 0; n < bin->export_directory->NumberOfNames; n++) {
-					function_ordinal = r_read_at_ble16 ((ut8 *)ordinals, n * sizeof (PE_Word), bin->endian);
+					PE_Word fo = r_read_at_ble16 ((ut8 *)ordinals, n * sizeof (PE_Word), bin->endian);
 					// if exist this index into AddressOfOrdinals
-					if (i == function_ordinal) {
+					if (i == fo) {
+						function_ordinal = fo;
```

Comment:
is this the real fix for the binary?

Reply:
no. this is a fix for another bug. the ordinal was assigned even if not found before this change. i think it is safer this way.